src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,132 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import duke.exceptions.IncompleteInputException;
+import duke.exceptions.InvalidInputException;
+import duke.exceptions.WrongMarkException;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+
+/**
+ * Represents a Parser object that deals with making sense of the user command.
+ */
+public class Parser {
+    /**
+     * Creates a Parser object.
+     */
+    public Parser() {
+    }
+
+    /**
+     * Checks the type of command given by the user.
+     * Executes the command given by the user.
+     *
+     * @param commandGiven The command given by the user.
+     * @param tasks        The TaskList object containing the tasks.
+     * @param ui           The Ui object to deal with user input and output.
+     * @param storage      The Storage object to deal with loading tasks from the file and saving tasks in the file.
+     * @return The String containing the output of the command.
+     * @throws IncompleteInputException If the command given is incomplete.
+     * @throws WrongMarkException       If the command given is wrong.
+     * @throws InvalidInputException    If the command given is invalid.
+     */
+    public static String checkCommandType(String commandGiven,
+                                          TaskList tasks,
+                                          Ui ui,
+                                          Storage storage)
+            throws IncompleteInputException,
+            WrongMarkException,
+            InvalidInputException {
+        commandGiven = commandGiven.trim();
+        String[] splittedCommands = commandGiven.split(" ");
+        String commandType = splittedCommands[0];
+        String command = "";
+        switch (commandType) {
+        case "list":
+            ui.separatorLines();
+            command = tasks.printTasks();
+            break;
+        case "mark":
+            try {
+                Task task = tasks.get(Integer.parseInt(splittedCommands[1]) - 1);
+                if (Objects.equals(task.getStatusIcon(), "X")) {
```

Comment:
Great to see you have added a functionality for this kind of actions, gives me ideas for my project as well

Reply:
I'm not sure which options are you referring to here, hehe